### PR TITLE
ref(settings): add search bar to repositories page

### PR DESCRIPTION
**Pull request by michellewzhang**
closes https://github.com/getsentry/sentry/issues/55836

SCR-20241212-noso

https://github.com/user-attachments/assets/8e1990ca-2757-4250-9077-cbf02079a179

### CODE REVIEW

**Review by ryan953 — APPROVED** [review]

File: static/app/views/settings/organizationRepositories/organizationRepositories.tsx [hunk 1]
```diff
@@ -33,6 +42,16 @@ function OrganizationRepositories({itemList, onRepositoryChange, organization}:
           'Want to add a repository to start tracking commits? Install or configure your version control integration here.'
         )}
       </AlertLink>
+      <StyledSearchBar>
+        <InputGroup.LeadingItems disablePointerEvents>
+          <IconSearch color="subText" size="xs" />
+        </InputGroup.LeadingItems>
+        <InputGroup.Input
+          ref={searchInput}
+          placeholder="Search repositories"
```
Comment (ryan953) [inline static/app/views/settings/organizationRepositories/organizationRepositories.tsx:51]: translation!
```suggestion
          placeholder={t("Search repositories")}
```

**Comment by scttcper** [comment]
feels like you would still need to loop through all the pages to find something?

**Comment by michellewzhang** [comment]
> feels like you would still need to loop through all the pages to find something?

what do you mean?

**Comment by scttcper** [comment]
this search bar only filters results on your current page and you would need to keep flipping pages to find the repo. in the ticket they're saying they have multiple pages

**Comment by michellewzhang** [comment]
> this search bar only filters results on your current page and you would need to keep flipping pages to find the repo. in the ticket they're saying they have multiple pages

ohh okay i see. i guess with the pagination implementation it makes it hard to search all of it at once. i'll close this for now since it doesn't seem to achieve what the customer is asking

**Pull request closed without merging**